operators/src/upgrade/rest/service.rs

```diff
@@ -62,21 +81,84 @@ impl Display for RestError {
 
 /// Put request for upgrade.
 #[put("/upgrade")]
-async fn apply_upgrade() -> Result<HttpResponse, RestError> {
-    // TODO: Start upgrade controller
-    let response = "success".to_string();
-    Ok(HttpResponse::Ok()
-        .content_type(ContentType::json())
-        .body(response))
+pub async fn apply_upgrade() -> Result<HttpResponse, RestError> {
+    match UpgradeOperatorConfig::get_config()
+        .k8s_client()
+        .create_upgrade_action_crd()
+        .await
+    {
+        Ok(()) => {
+            info!("UpgradeAction CRD created");
+        }
+        Err(error) => {
+            error!(?error, "failed to create UpgradeAction CRD");
+            let err =
+                RestError::default().with_error("unable to create UpgradeAction crd".to_string());
+            return Err(err);
+        }
+    }
+
+    match UpgradeOperatorConfig::get_config()
+        .k8s_client()
+        .create_upgrade_action_resource()
+        .await
+    {
+        Ok(u) => {
+            info!(
+                "Created UpgradeAction CustomResource '{}' in '{}' namespace.",
+                u.metadata.name.as_ref().unwrap(),
+                u.metadata.namespace.as_ref().unwrap()
+            );
```

Comment:
```suggestion
            info!(
               name = u.metadata.name.as_ref().unwrap(),
               namespace = u.metadata.namespace.as_ref().unwrap(),
                "Created UpgradeAction CustomResource"
            );
```
Also can we use name and namespace _any() ?

Reply:
Not sure what you mean by _any() @tiagolobocastro. 
Added suggestion to recent changes.

Reply:
If u has the trait `ResourceExt` then it has the method "name_any" and "namespace", so we'd be using getters rather than the fields directly.